pkg/cmd/cli/secrets/satoken.go

```diff
@@ -0,0 +1,121 @@
+
+	extractSATokenShort = `Extract the token from mountable secrets assigned to a ServiceAccount.`
+
+	extractSATokenLong = `
+Extract the token from mountable secrets assigned to a ServiceAccount.
+
+ServiceAccount API tokens are used by ServiceAccounts for privileged access to the API. If actions that cannot
+be accomplished by OpenShift client actions using the ServiceAccount are required, and the actions necessiate 
+privileged requests to the API server, the ServiceAccount token is necessary for authorization. This command will
+extract the ServiceAccount token from a ServiceAccount to facilitate these external actions.
+`
+
+	extractSATokenUsage = `%s SA-NAME`
+
+	extractSATokenExamples = `  # Extract the ServiceAccount token from ServiceAccount 'custom-sa'
+  $ %[1]s SA-NAME
+`
+)
+
+type ExtractSATokenOptions struct {
+	SAName        string
+	Namespace     string
+	SAClient      unversioned.ServiceAccountsInterface
+	SecretsClient unversioned.SecretsInterface
+
+	Out io.Writer
+	Err io.Writer
+}
+
+func NewCommandExtractSAToken(name, fullname string, f *cmdutil.Factory, out io.Writer) *cobra.Command {
+	options := &ExtractSATokenOptions{
+		Out: out,
+		Err: os.Stderr,
+	}
+
+	extractSATokenCommand := &cobra.Command{
+		Use:     fmt.Sprintf(extractSATokenUsage, name),
+		Short:   extractSATokenShort,
+		Long:    extractSATokenLong,
+		Example: fmt.Sprintf(extractSATokenExamples, fullname),
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := options.Complete(args, f); err != nil {
+				cmdutil.CheckErr(cmdutil.UsageError(cmd, err.Error()))
+			}
+
+			cmdutil.CheckErr(options.Run())
+		},
+	}
+
+	extractSATokenCommand.Flags().StringVarP(&options.Namespace, "namespace", "n", "", "the namespace that contains the ServiceAccount")
+	return extractSATokenCommand
+}
+
+func (o *ExtractSATokenOptions) Complete(args []string, f *cmdutil.Factory) error {
+	if len(args) != 1 {
+		return fmt.Errorf("expected one ServiceAccount name as an argument, got %q", args)
+	}
+
+	o.SAName = args[0]
+
+	client, err := f.Client()
+	if err != nil {
+		return err
+	}
+
+	o.SAClient = client.ServiceAccounts(o.Namespace)
+	o.SecretsClient = client.Secrets(o.Namespace)
+	return nil
+}
+
+func (o *ExtractSATokenOptions) Run() error {
+	sa, err := o.SAClient.Get(o.SAName)
+	if err != nil {
+		return fmt.Errorf("could not locate ServiceAccount %q: %v", o.SAName, err)
+	}
+
+	for _, reference := range sa.Secrets {
+		secret, err := o.SecretsClient.Get(reference.Name)
+		if err != nil {
+			continue
+		}
+
+		if secret.Type == kapi.SecretTypeServiceAccountToken {
+			token, exists := secret.Data[kapi.ServiceAccountTokenKey]
+			if !exists {
+				return fmt.Errorf("ServiceAccount token %q for ServiceAccount %q did not contain token data", secret.Name, sa.Name)
+			}
+
+			encoding := base64.StdEncoding
+			decodedToken := make([]byte, encoding.DecodedLen(len(token)))
+			length, err = encoding.Decode(decodedToken, token)
```

Comment:
You don't need to decode... just echo the data as `string(token)`. The base64 encoding is in the serialized API version when serializing `[]byte`